src/mlpack/methods/ann/ffn.hpp

```diff
@@ -177,6 +186,20 @@ class FFN
   //! Modify the initial point for the optimization.
   arma::mat& Parameters() { return parameter; }
 
+  /**
+   * Make the network ready for the given training data.
+   * This function won't actually trigger training process.
+   *
+   * @param predictors Input training variables.
+   * @param responses Outputs results from input training variables.
+   */
+  void SetTrainingData(const arma::mat& predictors, const arma::mat& responses);
```

Comment:
Do you think instead of using ``SetTrainingData`` followed by ``Gradient``we could provide another ``Gradient`` function, that takes the ``predictors`` and ``responses``?